src/components/Checkout/Checkout.js

```diff
@@ -0,0 +1,487 @@
+            response
+          );
+          console.log(data);
+        } catch (error) {
+          console.log(error);
+        }
+      },
+      theme: {
+        color: "green",
+      },
+      redirect: true,
+      callback_url: "http://localhost:3000/confirmation",
+    };
+    const rzp1 = new window.Razorpay(options);
+    rzp1.open();
+  };
+
+  useEffect(() => {
+    superCoinAPI();
+  }, []);
+
+  useEffect(() => {
+    getAllAddressAPI();
+  }, [oneAddress]);
+
+  return (
+    <>
+      <div className="checkout">
+        <div className="back">
+          <Link to="/cart">
+            <button>
+              <ArrowLeftOutlined /> Back
+            </button>
+          </Link>
+        </div>
+        <Steps className="steps">
+          <Step status="finish" title="Cart" icon={<ShoppingCartOutlined />} />
+          <Step
+            status="process"
+            title="Payment"
+            icon={<AlipayCircleOutlined />}
+          />
+          <Step
+            status="wait"
+            title="Confirmation"
+            icon={<CheckCircleOutlined />}
+          />
+        </Steps>
+        <div className="flexFilter">
+          <div>
+            <div className="addressCard">
+              <h1>Address</h1>
+              <div className="locationCard">
+                {allAddress.length === 0 ? (
+                  <div className="noAddress">No Saved Address</div>
+                ) : (
+                  allAddress.map((add) => (
+                    <div className={`savedCard ${newCardClass}`} key={add._id}>
+                      <h2>
+                        {add.firstName} {add.lastName}
+                      </h2>
+                      <p>
+                        {add.address} {add.apartmentNo} {add.city} {add.state}
+                        {add.pincode}
+                      </p>
+                      <button
+                        className={toggleClass}
+                        onClick={() => selectClick()}
+                      >
+                        {toggleButton ? "Selected" : "Select"}
+                      </button>
+                      <DeleteOutlined
+                        className="deleteIcon"
+                        onClick={() => deleteOneAddressAPI(add._id)}
+                      />
+                    </div>
+                  ))
+                )}
+              </div>
+            </div>
+
+            <div className="leadDetails">
+              <h2>or Add new address</h2>
+              <Form
+                autoComplete="off"
+                onFinish={(values) => {
+                  if (allAddress.length === 3) {
+                    moreThenThreeAddress();
+                  } else {
+                    onAddressSaveAPI(values);
+                  }
+                }}
+                form={form}
+              >
+                <div className="fullName">
+                  <Form.Item
+                    name="firstName"
+                    rules={[
```

Comment:
const rules = {[
                      {
                        required: true,
                        message: "Please enter First Name",
                      },
                      {
                        whitespace: true,
                        message: "First Name cannot be Empty",
                      },
                      {
                        min: 3,
                        message: "First Name must be Minimum 3 characters",
                      },
                      {
                        max: 15,
                        message: "First Name must be Under 15 characters",
                      },
                      {
                        type: Number,
                        message: "Number Not Allowed in First Name",
                      },
                    ]}
                    
                    
                    
                    
                    //////
                                      <Form.Item  rules=rules

Reply:
Done